api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -175,4 +220,53 @@ protected Obs deproxyResult(Obs result) {
 		obs.setConcept(deproxyObject(obs.getConcept()));
 		return obs;
 	}
+	
+	int getMax(SearchParameterMap theParams) {
+		return ((NumberParam) theParams.getParameters(FhirConstants.MAX_SEARCH_HANDLER).get(0).getParam()).getValue()
+		        .intValue();
+		
+	}
+	
+	protected List<String> handleRanking(List<Object[]> list, int max) {
```

Comment:
Can we have a comment here on the algorithm you're using, i.e., I find this code a little opaque.